lfs/gitscanner_log.go

```diff
@@ -216,7 +217,7 @@ type logScanner struct {
 // r: a stream of output from git log with at least logLfsSearchArgs specified
 func newLogScanner(dir LogDiffDirection, r io.Reader) *logScanner {
 	return &logScanner{
-		s:                   bufio.NewScanner(r),
+		r:                   bufio.NewReaderSize(r, 1024),
```

Comment:
I needs to be long enough to match the expressions, which I initially assumed was just hashes.  However now I see it's matching file names, so perhaps there's not really a safe way to do this without allocating memory and handling unlimited line lengths?

Reply:
Hmm, that's a good question, indeed.  Perhaps `bufio.ReadString()` should be our friend here instead of `ReadLine()`.  What about something like this?
```go
func (s *logScanner) scan() (*WrappedPointer, bool) {
	for {
		line, err := s.r.ReadString('\n')
		if err != nil && err != io.EOF {
			s.err = err
			return nil, false
		}

		// remove trailing newline delimiter and optional single carriage return
		line = strings.TrimSuffix(strings.TrimRight(line, "\n"), "\r")

		if match := s.commitHeaderRegex.FindStringSubmatch(line); match != nil {
		// existing regex if/else blocks
		}

		if err == io.EOF {
			break
		}
	}
```

The typical case of an empty `line` when `err == io.EOF` could be handled separately before the regular expressions are applied, I suppose, but it adds clutter and I'm not sure it gains much as the regexps should all fail on an empty string.  I think this as written would also cover the alternate case where EOF is returned along with a non-empty `line` string.

Reply:
I made this change and it does seem to work as expected.  

I'm still very new to go, and I don't fully understand the use of TrimRight and TrimSuffix.  From the docs:

> TrimSuffix returns s without the provided trailing suffix string. If s doesn't end with suffix, s is returned unchanged. 

> TrimRight returns a slice of the string s, with all trailing Unicode code points contained in cutset removed. 

To me that sounds like the former returns a copy of the string and the latter returns a slice.

My concern is only that it's reallocating and copying the string unnecessarily.

Reply:
That's a good thing to care about, indeed -- we wouldn't want to performing excess copies of long strings.

I believe `TrimSuffix()` will also just return a slice of the original, based on the [implementation](https://github.com/golang/go/blob/e9e0d1ef704c4bba3927522be86937164a61100c/src/strings/strings.go#L913) which just does:
```go
return s[:len(s)-len(suffix)]
```

So I think we don't have to worry about a secret copy being made here, but it's an excellent thing to have checked.